### PR TITLE
5th homework

### PR DESCRIPTION
**Pull request by Arcimiendar**
i'd like to merge my homework!

### CODE REVIEW

File: main/dz5/HTTPtask.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    if data.status_code == 404:
+        raise ValueError(f"Currency '{currency_abbreviation}' not found")
+
+    rate = data.json()["Cur_OfficialRate"]
+    scale = data.json()["Cur_Scale"]
+
+    return 1 / rate * scale
+
+
+def get_weather():
```
Comment (dmitry-mavchun) [inline main/dz5/HTTPtask.py]: missing type hint for return statement

File: main/dz5/HTTPtask.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
+from requests import get
+import re
+
+
+def exchange_rates(currency_abbreviation: str) -> float:
+
+    data = get(f"http://www.nbrb.by/API/ExRates/Rates/{currency_abbreviation}?ParamMode=2")
+
+    if data.status_code == 404:
```
Comment (dmitry-mavchun) [inline main/dz5/HTTPtask.py]: function will fail for any other 3xx, 4xx, 5xx code. so it's better to verify 200 code in this case I think.

File: main/dz5/HTTPtask.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    pattern_page = r"<a class=\"link link_theme_normal " \
+              r"place-list__item-name i-bem\" tabindex=\"0\" href=\"" \
+              r"(?P<ref>/pogoda/[a-z-]*)\" (?:[\w-]*)='{\"link\":{}}'>" \
+              r"(?P<name>[А-ЯЁ][ёа-я]+(?:[ -][ЁА-Я][ёа-я]+)?)</a>"
+
+    pattern_temperature = r"(?<=<div class=\"temp fact__temp fact__temp_size_s\"><span class=\"temp_" \
+                          r"_value\">)(?:\+)?(?P<temprt>[−]?[0-9]+)(?=</span><span class=\"temp__unit" \
+                          r" i-font i-font_face_yandex-sans-text-medium\">°</span></div>)"
+    data = get("https://yandex.by/pogoda/region/149")
+    cities = re.findall(pattern_page, data.text)
```
Comment (dmitry-mavchun) [inline main/dz5/HTTPtask.py]: regex is fine but why would you use `html` module from` lxml` package for html parsing

Comment (Arcimiendar) [reply main/dz5/HTTPtask.py]: i don't actually know, but i did it with lxml just in case)

**Pull request merged**